### DISCUSSION TITLE
Zasób umożliwiający pobranie aktualnej tabeli prowizji

### DISCUSSION THREAD

**Discussion by pgasidlo** [allegro/allegro-api#5369] [Suggestion - verification]
Szukałem, ale nie znalazłem w aktualnym REST API.

Potrzebuje zasobu, który na bieżąco pozwalałby mi pobierać jaka jest prowizja w określonej kategorii w zależności od ceny "kup teraz". Chciałbym to wiedzieć przed wystawieniem aukcji.

Aktualne dane są w formie "dla człowieka", ale nie dla "maszyny":

https://allegro.pl/regulamin/zalacznik/4#czesc-iii-prowizje-od-sprzedazy-dla-kont-firma

Prosty przypadek: w kategorii X mamy prowizję dla normalnej aukcji %A niezależnie od ceny "kup teraz" w oferty.

```
[
  ...
  {
    "id": X, // kategoria, której dotyczy wpis (i jej podkategorie o ile nie zostaną osobno zdefiniowane)
    "commissionPct": A, // prowizja w %
    "commissionAbsolute": 0.00, // wyjaśnione niżej
    "priceGt": 0.00, // cena od
    "priceLe": null, // cena do, null - bez limitu
  }
  ...
]
```

Skomplikowany przypadek: w kategorii X  mamy prowizję dla normalnej aukcji %A a dla wyróżnionej %B, ale dotyczy to ofert, gdzie cena "kup teraz" jest większa niż MIN ale mniejsza niz MAX.

Przykładowo:

"Dziecko" (podkategoria "Zabawki" z wyłączeniem podkategorii "Klocki", "Gry elektroniczne", "Słowne i liczbowe", "Zabawki zręcznościowe"):

`Wysokość prowizji zależy od ceny końcowej: do 50 zł: 14%,powyżej 50 zł: 7 zł + 8% od nadwyżki ponad 50 zł`

Wtedy:
```
[
  ...
  {
    "id": 11818,
    "commissionPct": 14.00,
    "commissionAbsolute": 0.00, // wyjaśnione niżej i w przykładzie
    "priceGt": 0.00,
    "priceLe": 50.00,
  },
  {
    "id": 11818,
    "commissionPct": 8.00,
    "commissionAbsolute": 7.00, // bezwględna prowizja, którą należy dodać do wyliczeń
    "priceGt": 50.00,
    "priceLe": null, // bez limitu
  },
  ... // dodatkowe rekordy opisujące reguły dla "Klocki", "Gry elektroniczne", "Słowne i liczbowe", "Zabawki zręcznościowe"
]
```

Przykładowo, dla produktu w cenie 90.00 zł. Znajdujemy, pasujący rekord (id, priceGt < 90.00 <= priceLe):

`prowizja bewzględna = (90.00 - priceGt) * commissionPct% + commissionAbsolute = (90.00 - 50.00) * 0.08 + 7.00 = 10.20`

Jeżeli w API nie są podawane dane odnośnie jakiejś kategorii należy sprawdzać kolejno rekord dla kolejnych kategorii nadrzędnych, aż do momentu "trafienia".

**Comment by pliszka** [comment]
@pgasidlo na razie wydaje się że obejściem tego problemu może być cykliczne odpytywanie z wszystkimi danymi oferty: https://developer.allegro.pl/documentation/#operation/calculateFeePreviewUsingPOST
